Plugins/Public/playercntl_plugin/MiscCmds.cpp

```diff
@@ -252,6 +263,185 @@ namespace MiscCmds
 		return true;
 	}
 
+	void MiscCmds::LoadListOfReps()
+	{
+		INI_Reader ini;
+
+		string factionpropfile = "..\\data\\initialworld.ini";
+		if (ini.open(factionpropfile.c_str(), false))
+		{
+			while (ini.read_header())
+			{
+				if (ini.is_header("Group"))
+				{
+					uint ids_name;
+					string nickname;
+					while (ini.read_value())
+					{
+						if (ini.is_value("nickname"))
+						{
+							nickname = ini.get_value_string();
+						}
+						else if (ini.is_value("ids_name"))
+						{
+							ids_name = ini.get_value_int(0);
+						}
+
+					}
+					factions[nickname] = ids_name;
+				}
+			}
+			ini.close();
+			ConPrint(L"Resetrep: Loaded %u factions\n", factions.size());
+		}
+	}
+	
+	map<wstring, uint> MiscCmds::Resetrep_Time_limits_load_map(string filename)
+	{
+		INI_Reader ini;
+		string factionpropfile = filename;
+		map<wstring, uint> tempmap;
+		if (ini.open(factionpropfile.c_str(), false))
+		{
+			while (ini.read_header())
+			{
+				if (ini.is_header("Playership"))
+				{
+					uint timelimit = 0;
+					wstring nickname = L"0";
+					while (ini.read_value())
+					{
+						if (ini.is_value("shipname"))
+						{
+							ini_get_wstring(ini, nickname);
+						}
+						else if (ini.is_value("timelimit"))
+						{
+							timelimit = ini.get_value_int(0);
+						}
+					}
+
+					tempmap[nickname] = timelimit;
+				}
+			}
+			ini.close();
+		}
+		return tempmap;
+	}
+
+	void MiscCmds::Resetrep_Time_limits_add(string filename, map<wstring, uint> tempmap)
+	{
+		string factionpropfile = filename;
+
+		FILE *file = fopen(factionpropfile.c_str(), "w");
+		if (file)
+		{
+			for (map<wstring, uint>::iterator i = tempmap.begin();
+				i != tempmap.end(); ++i)
+			{
+				if (((int)time(0) - i->second) < set_iResetrepTimeLimit)
```

Comment:
wait why is this here??

Reply:
if only I could remember...

Reply:
from what I was able to find... in the code which I certainly do not remember anymore

those two extra functions were made as 'wrappers' to deal with time limits being stored in files.

it looks to be very useful when 'data' about time limits needs to be 'loaded' again in rename and move char functions
![image](https://user-images.githubusercontent.com/20555918/104808274-9f989080-5817-11eb-8281-4b507de13ac1.png)

Reply:
more precide namings for those functions would be
Resetrep_load_time_limit_for_player account(folder)
Resetrep_save_time_limits_for_player_account(folder)

Reply:
After re-reading this code, I still think those functions are still needed. Because I don't want to write fully code to 'read' data from files for all three times it is needed

Reply:
I could make it more looking like a one class wrapping those dealings though.

something like
class time_limits_for_player_account
{
map_of_time_limits
construct()
{ load from file }
save()
{ save to file }
get_time_limits()
{ return map_of_time_limits }
}

Reply:
Resetrep_save_time_limits_for_player_account is used three times, good enough to be a function and not copy pasted code

Reply:
commited and squashed renames to functions

Reply:
I still don't understand why we're checking the current time inside a function intended to write a map to disk.

Reply:
This was intended... to write to disk only time limits which have not expired yet
nvm, answering in github